halo2_proofs/src/poly/domain.rs

```diff
@@ -360,6 +385,27 @@ impl<F: WithSmallOrderMulGroup<3>> EvaluationDomain<F> {
         });
     }
 
+    fn fft_inner(&self, a: &mut Vec<F>, omega: F, log_n: u32, inverse: bool) {
+        let start = get_time();
+        if get_fft_mode() == 1 {
+            let fft_data = if a.len() == self.fft_data.n {
+                &self.fft_data
+            } else {
+                &self.extended_fft_data
+            };
+            recursive::fft(a, omega, log_n, fft_data, inverse);
+        } else {
+            // TODO
+            best_fft(a, omega, log_n, &FFTData::default(), false);
+        }
```

Comment:
```suggestion
        let fft_data = if a.len() == self.fft_data.n {
            &self.fft_data
        } else {
            &self.extended_fft_data
        };
        best_fft(a, omega, log_n, fft_data, false);
```

Should use the selected FFT everywhere.  `get_fft_mode` shouldn't be used anymore as your `var("FFT")` env variable is now used instead.